feat: Update collection sorting, metadata, stats

tw4l opened the pull request:
Fixes #2326 

This slightly reworks the existing page counts in our API to include `snapshotCount` as distinct from `pageCount` (which now only counts unique URLs).

Tweaks migration 0040 (will require `rerun_from_migration` to be set to rerun this on dev or elsewhere where we've run that already) to include snapshot count and adds a new migration 0041 to update collection counts.

tw4l committed:
Add snapshotCount to archived items

tw4l committed:
Rationalize counts for collections, add migration to recalc

tw4l committed:
Add snapshotCounts to org metrics endpoint

tw4l committed:
Update tests

tw4l committed:
Recalculate collection counts on import

tw4l committed:
Fix linting

tw4l committed:
Add pylint comment

tw4l committed:
Fix collection test

SuaYoo commented:
@tw4l looks like we also need to decrement the counts when archived items are deleted.

tw4l committed:
Ensure page counts are based on unique URLs in that object specifically

tw4l committed:
Add pylint comment

tw4l committed:
Update collections when archived items are deleted

tw4l committed:
Add pylint comment

tw4l commented:
> @tw4l looks like we also need to decrement the counts when archived items are deleted.

Done, good catch! Looks like that was a bug that's been around for a while.

tw4l commented:
@ikreymer marked this as ready for review. I've fixed the issues Sua found while testing and tested locally that all looks good, but we decided to do the frontend updates in a separate PR.

tw4l committed:
Add missing await

SuaYoo commented:
Tested locally with https://github.com/webrecorder/browsertrix/pull/2324, LGTM!

SuaYoo committed:
feat: Update collection sorting & metadata (#2324)

- Fixes https://github.com/webrecorder/browsertrix/issues/2321
- Resolves https://github.com/webrecorder/browsertrix/issues/2323

Follows https://github.com/webrecorder/browsertrix/pull/2327, should be
rebased and merged afterwards

## Changes

- Refactors dashboard and org profile preview to use private API
endpoint, to fix public collections not showing the org visibility is
hidden
- Enables sorting collections by `dateLatest`, sorts public collections
by `dateLatest` by default
- Enables sorting collections by page count
- Shows collection period (i.e. `dateEarliest` to `dateLatest`) in
collections list
- Shows same collection metadata in private and public views, updates
private view info bar
- Fixes "Update Org Profile" action item showing for crawler roles





<!-- ## Follow-ups -->

---------

Co-authored-by: Tessa Walsh <tessa@bitarchivist.net>

ikreymer commented:
Per suggestions in discord, I think we should keep `pageCount` as is and add a `uniquePageCount`, and not use the term snapshot at all..

I think term snapshot could easily be confused for non-page URLs as well as screenshots..

tw4l committed:
Change backend terminology: use pageCount and uniquePageCount

SuaYoo committed:
update field

SuaYoo committed:
replace instances of snapshot

SuaYoo commented:
@ikreymer @tw4l Updated frontend, not sure if it's a migration issue but `uniquePageCount` is returning `0` even though there are pages crawled (running the backend locally):
<img width="195" alt="Screenshot 2025-01-22 at 1 24 40 PM" src="https://github.com/user-attachments/assets/33ab7c32-85fc-4f3c-b367-ad23b766021a" />

Also updated this dropdown (cc @Shrinks99 )
<img width="469" alt="Screenshot 2025-01-22 at 1 27 02 PM" src="https://github.com/user-attachments/assets/9573c812-c0ec-45bf-b9a3-1366fbd9f197" />

SuaYoo committed:
update total pages

SuaYoo committed:
fix wrong count used

SuaYoo committed:
update strings

SuaYoo reviewed (approved):
Tested both backend and frontend locally, looks good to me

ikreymer commented on backend/btrixcloud/models.py:
```diff
@@ -1919,6 +1924,9 @@ class OrgMetrics(BaseModel):
     crawlPageCount: int
     uploadPageCount: int
+    uniquePageCount: int
+    crawlUniquePageCount: int
```
Shouldn't crawls always have unique pages?

ikreymer replied:
ah nevermind, that's on the org!

tw4l replied:
yes and also uploads can have multiple captures of the same URL so can have different numbers within an archived item as well

ikreymer committed:
fix typo in migration

ikreymer reviewed (approved):
fixed small typo in migration, should be good now!

ikreymer committed:
test: add crawlUniquePageCount and uploadUniquePageCount to tests
